### PR TITLE
bug fix to avoid nil pointer

### PR DESCRIPTION
**Pull request by rhsakarpos**
**What this PR does / why we need it**:
This fixes a nil pointer bug when the bucket does NOT have any SSE settings

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by rhsakarpos** [comment]
@kumarashit @nguptaopensds raised this fix separately as discussed, to be merged to continue testing from dev. branch. @nguptaopensds can you please confirm from your env. that this is indeed the issue due to which you are unable to upload the object ? thanks.

**Comment by kumarashit** [comment]
@nguptaopensds Can you please update the test result?

**Comment by sfzeng** [comment]
LGTM

File: api/pkg/s3/listbuckets.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ func parseListBuckets(list *s3.ListBucketsResponse) []byte {
 	for _, value := range list.Buckets {
 		ctime := time.Unix(value.CreateTime, 0).Format(time.RFC3339)
 		versionOpts := model.VersioningConfiguration{}
+		versionOpts.Status = utils.VersioningDisabled
```
Comment (sfzeng) [inline api/pkg/s3/listbuckets.go]: Hi, why always 'disabled' here?

File: api/pkg/s3/listbuckets.go [hunk 2]
```diff
@@ -50,9 +51,11 @@ func parseListBuckets(list *s3.ListBucketsResponse) []byte {
 		if value.ServerSideEncryption != nil {
 			if value.ServerSideEncryption.SseType == "SSE" {
 				sseOpts.SSE.Enabled = "true"
+			} else {
+				sseOpts.SSE.Enabled = "false"
 			}
 		}
-		bucket := model.Bucket{Name: value.Name, CreateTime: ctime, LocationConstraint: value.DefaultLocation, SSEOpts: sseOpts}
+		bucket := model.Bucket{Name: value.Name, CreateTime: ctime, LocationConstraint: value.DefaultLocation, VersionOpts:versionOpts, SSEOpts: sseOpts}
```
Comment (sfzeng) [inline api/pkg/s3/listbuckets.go]: Format code please.

Comment (rhsakarpos) [reply api/pkg/s3/listbuckets.go]: done

File: api/pkg/s3/listbuckets.go (hunk 1 above)
Comment (rhsakarpos) [reply api/pkg/s3/listbuckets.go]: default is disabled. then it gets set to whatever the option on the bucket is, if any

**Pull request merged**